machinelearning/image_metadata_to_JSON.go

```diff
@@ -201,37 +201,35 @@ func ImageMetadataToJSONLines(matchingData []*ImageMetadata,
 				}
 
 			case mlv1.ModelType_MODEL_TYPE_UNSPECIFIED:
-				return errors.New("model type not specified")
+				return nil, 0, errors.New("model type not specified")
 			}
 		}
 
 		line, err := json.Marshal(jsonl)
 		if err != nil {
-			return errors.Wrap(ErrJSONFormatting, err.Error())
+			return nil, 0, errors.Wrap(ErrJSONFormatting, err.Error())
 		}
 		line = append(line, "\n"...)
 		_, err = wc.Write(line)
 		if err != nil {
-			return errors.Wrap(ErrFileWriting, err.Error())
+			return nil, 0, errors.Wrap(ErrFileWriting, err.Error())
 		}
 	}
 
 	// For non-custom training, perform validation on the dataset.
 	if requestedTags != nil {
-		// TODO(DATA-1541): Use DB queries for ML training data validations and move to SubmitTrainingJob
-		if tooManyLabels == len(matchingData) {
```

Comment:
Anything that fails this check will return the `errTooFewAnnotations` error, so I'm not sure it's necessary. We would require another DB query to recreate this in `ValidateTrainingJobWithCounts` so I prefer removing.